### PR TITLE
PLAT-126504/Document-enhancements-to-the-Sandbox-API-Reference [REVIEW]

### PR DESCRIPTION
**Pull request by hbhandal80**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: src/swagger-specs/sandbox.yaml [hunk 1]
```diff
@@ -458,79 +460,107 @@ definitions:
...
       lastModifiedDate:
         type: string
+        description: The date and time the sandbox was last modified.
         format: 'date-time'
       createdBy:
         type: string
+        description: Details of the individual who created the sandbox.
       modifiedBy:
         type: string
+        description: Details of the individual who last modified the sandbox.
```
Comment (JCON3DEV) [inline src/swagger-specs/sandbox.yaml:563]: same as above, needs an example

File: src/swagger-specs/sandbox.yaml [hunk 2]
```diff
@@ -546,37 +576,51 @@ definitions:
...
         type: string
+        description: The display name for the sandbox.
         example: My Sandbox
       state:
         type: string
+        description: The current processing state of the sandbox. The state of a sandbox can be any of the following:\n
+          - creating:\n The sandbox has been created, but is still being provisioned by the system.\n
+          - active:\n The sandbox is created and active.\n
+          - deleted:\n The sandbox has been manually disabled.\n
+          - failed:\n Due to an error, the sandbox was not able to be provisioned by the system and is disabled.\n
```
Comment (JCON3DEV) [inline src/swagger-specs/sandbox.yaml]: previous comments apply to repeated content.

File: src/swagger-specs/sandbox.yaml [hunk 3]
```diff
@@ -458,79 +460,107 @@ definitions:
...
+        description: An identifier for a specific version of the sandbox. Used for version control and caching efficiency.
         example: 2
       createdDate:
         type: string
+        description: The date and time the sandbox was created.
         format: 'date-time'
       lastModifiedDate:
         type: string
+        description: The date and time the sandbox was last modified.
         format: 'date-time'
```
Comment (JCON3DEV) [inline src/swagger-specs/sandbox.yaml:557]: format: 'date-time' seems to generate an example so none needed,..

File: src/swagger-specs/sandbox.yaml [hunk 4]
```diff
@@ -546,37 +576,51 @@ definitions:
         example: d290f1ee-6c54-4b01-90e6-d701748f0851
       name:
         type: string
+        description: The name of the sandbox in which the operation will take place.
```
Comment (JCON3DEV) [inline src/swagger-specs/sandbox.yaml:579]: lines 567-570 should be removed

File: src/swagger-specs/sandbox.yaml [hunk 5]
```diff
@@ -597,14 +641,18 @@ definitions:
         example: active
       createdDate:
         type: string
+        description: The date and time the sandbox was created.
         format: 'date-time'
       createdBy:
         type: string
+        description: Details of the individual who created the sandbox.
```
Comment (JCON3DEV) [inline src/swagger-specs/sandbox.yaml:648]: just a reminder to put in examples for last created and modified by.

File: src/swagger-specs/sandbox.yaml [hunk 6]
```diff
@@ -614,12 +662,15 @@ definitions:
         example: 180e1210-d33a-11e9-8b17-d9e538719ued
       name:
         type: string
+        description: The identifier that will be used to access the sandbox in future requests. This value must be unique, and best practice is to make it as descriptive as possible. This value cannot contain any spaces or special characters.
```
Comment (JCON3DEV) [inline src/swagger-specs/sandbox.yaml:665]: As this is human generated. It should be mentioned explicitly.  Although, this is a description for the response so guidance on name selection is unnecessary. it needs reframing as 'The user generated human friendly name for the sandbox that will be used ...'
The creation description should be in 'CreateSandboxPayload'
This then becomes a rabbit hole of fixing the swagger docs 😂 
also, this ref is poorly named, ha.

File: src/swagger-specs/sandbox.yaml [hunk 7]
```diff
@@ -614,12 +662,15 @@ definitions:
...
         example: mysandbox
       title:
         type: string
+        description: A human-readable name used for display purposes in the Platform user interface.
         example: My Sandbox
       state:
         type: string
+        description: Sandboxes take roughly 30 seconds to be provisioned by the system, after which their state will become “active” or “failed”.
```
Comment (JCON3DEV) [inline src/swagger-specs/sandbox.yaml:673]: we should backtick these values `active` `failed`
I would also reframe this slightly. It is: "The state of the sandbox and has two values, `active` `failed`. A sandbox becomes `active` once it has been successfully provisioned." or sth to that effect

File: src/swagger-specs/sandbox.yaml [hunk 8]
```diff
@@ -631,6 +682,7 @@ definitions:
         example: creating
       type:
         type: string
+        description: The type of sandbox to be created. For a non-production sandbox, this value must be development. For a production sandbox, this value must be production.
```
Comment (JCON3DEV) [inline src/swagger-specs/sandbox.yaml:685]: just backtick `development` and `production`.

File: src/swagger-specs/sandbox.yaml [hunk 9]
```diff
@@ -645,12 +697,14 @@ definitions:
...
       name:
         type: string
+        description: The name of the sandbox you want to delete.
         example: mysandbox
       title:
         type: string
         example: My Sandbox
       state:
         type: string
+        description: Deleting a sandbox updates the sandbox state property to “deleted” and deactivates it. The default production sandbox cannot be deleted.
```
Comment (JCON3DEV) [inline src/swagger-specs/sandbox.yaml:707]: I would add some backticks  (state and first mention of deleted), plus remove the word property:
"A successful deletion request a sandbox updates the sandbox `state` to `deleted` and deactivates it. The default production sandbox cannot be deleted."

File: src/swagger-specs/sandbox.yaml [hunk 10]
```diff
@@ -645,12 +697,14 @@ definitions:
         example: 180e1210-d33a-11e9-8b17-d9e538719ued
       name:
         type: string
+        description: The name of the sandbox you want to delete.
         example: mysandbox
       title:
```
Comment (JCON3DEV) [inline src/swagger-specs/sandbox.yaml:702]: This needs a description

**Review by JCON3DEV — COMMENTED** [review]
mostly its just backticks missing and some unnecessary required tags.

**Comment by calebk1** [comment]
retest

**Pull request merged**